src/libostree/ostree-repo.c

```diff
@@ -1060,6 +1090,11 @@ ostree_repo_finalize (GObject *object)
         g_private_replace (&repo_lock_table, NULL);
     }
 
+  g_mutex_lock (&self->lock.mutex);
```

Comment:
It looks like the rest of this function does not try lock the other mutexes, which I guess could be useful in case of a poisoned mutex due to a thread panic.
Is racing between a lock and a finalize operation (e.g. across different threads) a concern for the library?

Reply:
I don't think it is a concern. I was thinking about what would happen if you tried to finalize while another thread is still actively using the same object, but I think you'd have bigger problems if that was the case. I think I'll just drop this lock/unlock and assume that anyone using threads will clean them up before finalizing or deal with the consequences.